### DISCUSSION TITLE
EFR32MG21 as a Coordinator and Webee CC2530 + CC2591 as router - max number of devices

### DISCUSSION THREAD

**Discussion by hlawati14** [Koenkk/zigbee2mqtt#20781] [Devices]
Hello,

I am developing a Zigbee based smart system for a muli-floor building. Each floor has 10-20 rooms.
I am using the [Dusun DSGW-210 Computer](https://www.dusuniot.com/product-specification/iot-edge-computer-gateway-spc/) as my main controller and coordinator. It houses an EFR32MG21 Zigbee chip.

I [extended ](https://www.zigbee2mqtt.io/advanced/zigbee/05_create_a_cc2530_router.html#result) the zigbee network using Webee CC2530 + CC2591 Chip flashed with a [router firmware](https://github.com/Koenkk/Z-Stack-firmware/tree/master/router/Z-Stack_Home_1.2/bin).

I have seen many sources mentioning different information about the maximum number of devices I can pair, such as [this](https://www.zigbee2mqtt.io/guide/faq/#i-read-that-zigbee2mqtt-has-a-limit-of-20-devices-when-using-a-cc2530-cc2531-adapter-is-this-true)

I am planning to place a coordinator in each floor, and use routers in each room. All devices in a room would pair to their respective router.

My questions are
-1- How can I identify the max number of routers as well as end devices I can add to my network?
-2- Also, is it okay to "Daisy Chain" the routers?

Thanks

**Comment by JorgeMarron14** [comment]
Hi! I am also using a Dusun device, the DSGW-030-1, as a gateway. You are one of the few people I see who has posted information about the use of these devices. Could you tell me how to see logs on the device? I'm trying to pair an SNZB-03 and it's impossible, it never lists it in the IOT Services/Zigbee menu. I'm sorry I can't help you with your question and I hope you can solve it quickly!
